packages/client/hooks/useGroupMatrix.ts

```diff
@@ -0,0 +1,97 @@
+import {SpotlightGroupsQueryResponse} from './../__generated__/SpotlightGroupsQuery.graphql'
+import {RefObject, useEffect, useState} from 'react'
+import {ElementWidth} from '../types/constEnums'
+import useResizeObserver from './useResizeObserver'
+import {MAX_SPOTLIGHT_COLUMNS} from '~/utils/constants'
+import useInitialRender from './useInitialRender'
+
+type Group = SpotlightGroupsQueryResponse['viewer']['similarReflectionGroups'][0]
+
+const useGroupMatrix = (
+  resultsGroups: readonly Group[],
+  resultsRef: RefObject<HTMLDivElement>,
+  phaseRef: RefObject<HTMLDivElement>
+) => {
+  const isInit = useInitialRender()
+
+  const getColumnsCount = () => {
+    const {current: el} = resultsRef
+    const width = el?.clientWidth
+    if (!width) return null
+    let colCount = 1
+    const getNextWidth = (count: number) =>
+      ElementWidth.MEETING_CARD * count + ElementWidth.MEETING_CARD_MARGIN * (count - 1)
+    while (getNextWidth(colCount + 1) < width && colCount + 1 <= MAX_SPOTLIGHT_COLUMNS) {
+      colCount++
+    }
+    return colCount
+  }
+
+  const initMatrix = (): Group[][] => {
+    const columnsCount = getColumnsCount()
+    if (columnsCount === null) return []
+    const matrix = Array.from(new Array(columnsCount), () => []) as Group[][]
```

Comment:
Original suggestion was with `const matrix = Array(columnsCount).fill([]) as Group[][]`. I learnt something new as it turns out `Array.fill()` fills the array with a reference to the same array. `Array.from` with a callback has the desired behaviour: https://stackoverflow.com/questions/41121982/strange-behavior-of-an-array-filled-by-array-prototype-fill